aten/src/ATen/core/type.cpp

```diff
@@ -655,4 +655,90 @@ void checkNoAny(const Type& base, const char* what, const std::string& attrname,
       "' but it contains an Any type. Any types cannot be members of modules, classes, or named tuples.");
 }
 
+namespace {
```

Comment:
Since this is mobile specific we should put it in the mobile folder and name it something to indicate that it is the mobile version of type parser.